src/include/rados/librados.h

```diff
@@ -1453,6 +1453,22 @@ CEPH_RADOS_API int rados_read(rados_ioctx_t io, const char *oid, char *buf,
                               size_t len, uint64_t off);
 
 /**
+ * Compute checksum from object data
+ *
+ * The io context determines the snapshot to checksum, if any was set
+ * by rados_ioctx_snap_set_read().
+ *
+ * @param io the context in which to perform the checksum
+ * @param oid the name of the object to checksum
+ * @param len the number of bytes to checksum
+ * @param off the offset to start checksuming in the object
+ * @param pcrc where to store the checksum result
+ * @return negative error code on failure
+ */
+CEPH_RADOS_API int rados_checksum(rados_ioctx_t io, const char *oid,
+                                  size_t len, uint64_t off, uint32_t *pcrc);
```

Comment:
Fair enough -- I was just trying to get away w/ re-using the existing extent structure instead of rolling my own "checksum" struct with duplicated extents. :-p